entegywrapper/schemas/content.py

```diff
@@ -164,12 +164,50 @@ class Content(BaseModel):
     selectedCategories: Optional[list[Category]] = None
     children: Optional[list["Content"]] = None
 
+    def get_updated_content(self, other: "Content") -> dict[str, Any]:
+        """
+        Get a dictionary of updated content when comparing another content object against self.
+        The list of items that can be updated is guided by Entegy's API docs however a number of the
+        optional attributes have been left in. Time will tell if this is a valid assumption.
+
+        Return should really be an object. See:
+        https://github.com/SituDevelopment/entegy-sdk-python/issues/184
+
+        Parameters
+        ----------
+            `other` (`Content`): The content object against which to compare self.
+
+        Returns
+        -------
+            `dict[str, Any]`: The results dict. Could really use an object here. To be fixed in
+            https://github.com/SituDevelopment/entegy-sdk-python/issues/184
+        """
+        updated_content = {}
+        if self.name != other.name:
+            updated_content["name"] = other.name
+        if self.mainImage != other.mainImage:
+            updated_content["mainImage"] = other.mainImage
+        if self.strings != other.strings:
+            updated_content["strings"] = other.strings
+        if self.pageSettings != other.pageSettings:
+            updated_content["pageSettings"] = other.pageSettings
+        if self.sortOrder != other.sortOrder:
+            updated_content["sortOrder"] = other.sortOrder
+        if self.documents != other.documents:
+            updated_content["documents"] = other.documents
+        if self.links != other.links:
+            updated_content["links"] = other.links
+        if self.multiLinks != other.multiLinks:
+            updated_content["multiLinks"] = other.multiLinks
+        if self.selectedCategories != other.selectedCategories:
+            updated_content["selectedCategories"] = other.selectedCategories
+        return updated_content
```

Comment:
I think this should be more dynamic. Consider something like the below:

```suggestion
        updated_content = {}
        diff = lambda field: getattr(self, field) != getattr(other, field)

        for field in filter(diff, self.__class__.model_fields.keys()):
            updated_content[field] = getattr(other, field)

        return updated_content
```

or an equivalent comprehension:
        
```suggestion
        return {
            field: getattr(other, field)
            for field in self.__class__.model_fields
            if getattr(self, field) != getattr(other, field)
        }
```

See the [`BaseModel.model_fields` docs](https://docs.pydantic.dev/latest/api/base_model/#pydantic.BaseModel.model_fields).